### PR TITLE
community(you): Integrate You.com conversational APIs

### PR DESCRIPTION
**Pull request by christeefy**
You.com is releasing two new conversational APIs — Smart and Research. 

This PR:
- integrates those APIs with Langchain, as an LLM
- streaming is supported

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
these are cool! i think you need to merge master before CI can run

File: libs/community/tests/unit_tests/llms/test_you.py
Comment (christeefy) [inline libs/community/tests/unit_tests/llms/test_you.py:1]: I do not know how to use the `requests_mock` library to mock a stream of bytes. I've attempted to use `requests.post(..., content=content_callback)` where `content_callback` returns a whole `byte` string but that doesn't work. If I have force `content_callback` to return a `byte` generator, `requests_mock` complains.

**Comment by rmcarthur** [comment]
Checking in from You.com. We are hoping to rapidly increase and improve APIs. Let me know if we can support in any way.

**Comment by rmcarthur** [comment]
@hwchase17 how do we get this merged?

**Comment by christeefy** [comment]
@ccurme, since you previously reviewed a related #22622, could you help us to get this over the line? 
Thank you very much in advance!

**Review by ccurme — APPROVED** [review]

**Comment by christeefy** [comment]
Thanks @ccurme!

**Pull request merged**